server_environment/models/server_env_tech_name_mixin.py

```diff
@@ -16,56 +17,63 @@ class ServerEnvTechNameMixin(models.AbstractModel):
     This mixin helps solve the problem by providing a tech name field
     and a cleanup machinery as well as a unique constrain.
 
-    To use this mixin add it to the _inherit attr of your module like:
-
-        _inherit = [
-            "my.model",
-            "server.env.techname.mixin",
-            "server.env.mixin",
-        ]
-
+    Use it in place of "server.env.mixin".
     """
 
     _name = "server.env.techname.mixin"
+    _inherit = "server.env.mixin"
     _description = "Server environment technical name"
     _sql_constraints = [
         ("tech_name_uniq", "unique(tech_name)", "`tech_name` must be unique!",)
     ]
-    # TODO: could leverage the new option for computable / writable fields
-    # and get rid of some onchange / read / write code.
+
     tech_name = fields.Char(
+        string="Technical Name",
         help="Unique name for technical purposes. Eg: server env keys.",
+        compute="_compute_tech_name",
+        store=True,
+        readonly=False,
     )
 
     _server_env_section_name_field = "tech_name"
 
-    @api.onchange("name")
-    def _onchange_name_for_tech(self):
-        # Keep this specific name for the method to avoid possible overrides
-        # of existing `_onchange_name` methods
-        if self.name and not self.tech_name:
-            self.tech_name = self.name
+    @api.depends("name")
+    def _compute_tech_name(self):
+        for rec in self:
+            # Update tech_name only if it hasn't been set or if we're
+            # dealing with a new record.
+            if not rec.tech_name or not rec._origin.id:
+                rec.tech_name = self._normalize_tech_name(rec.name)
 
     @api.onchange("tech_name")
     def _onchange_tech_name(self):
-        if self.tech_name:
-            # make sure is normalized
-            self.tech_name = self._normalize_tech_name(self.tech_name)
-
-    @api.model
-    def create(self, vals):
-        self._handle_tech_name(vals)
-        return super().create(vals)
-
-    def write(self, vals):
-        self._handle_tech_name(vals)
-        return super().write(vals)
-
-    def _handle_tech_name(self, vals):
-        # make sure technical names are always there
-        if not vals.get("tech_name") and vals.get("name"):
-            vals["tech_name"] = self._normalize_tech_name(vals["name"])
+        # make sure it's normalized
+        res = {}
+        normalized = self._normalize_tech_name(self.tech_name)
+        if self.tech_name != normalized:
```

Comment:
I still feel that this useless and we can simply describe the behavior in the `help` attribute. But... yeah whatever :sweat_smile: